library/test_search.cc

```diff
@@ -96,18 +98,19 @@ void train()
 {
   // initialize VW as usual, but use 'hook' as the search_task
   cerr << endl << endl << "##### train() #####" << endl << endl;
-  VW::workspace& vw_obj =
-      *VW::initialize("--search 4 --quiet --search_task hook --example_queue_limit 1024 -f my_model");
-  run(vw_obj);
-  VW::finish(vw_obj, false);
+  auto vw_obj = VW::initialize(VW::make_unique<VW::config::options_cli>(
+      std::vector<std::string>{"--search 4 --quiet --search_task hook --example_queue_limit 1024 -f my_model"}));
```

Comment:
Needs to be split into individual strings or vw_split_command?

Reply:
Good catch thanks!